tests/validate/common/includes/xcl2/xcl2.cpp

```diff
@@ -52,6 +63,28 @@ std::vector<cl::Device> get_devices(const std::string &vendor_name) {
 
 std::vector<cl::Device> get_xil_devices() { return get_devices("Xilinx"); }
 
+cl::Device find_device_bdf(const std::string & bdf)
```

Comment:
I think this function should take the device list as first parameter to avoid calling get_xil_devices again, at least an overload should be provided:
```
cl::Device 
find_device_bdf(const std::vector<cl::Device>& devices, const std::string& bdf)
{
   ...
}
cl::Device 
find_device_bdf(const std::string& bdf)
{
  return find_device_bdf(xcl::get_xil_devices(), bdf);
}
```

Reply:
updated